src/menu.c

```diff
@@ -392,4 +392,90 @@ static gpointer context_menu_thread(gpointer data)
 
         return NULL;
 }
+
```

Comment:
Right now the new code in menu.c is not integrated with the old code. This could lead to confusion with other similar functions in this file. I would suggest to either
1. create a separate dmenu.c for all dmenu stuff and renaming those functions to reflect the fact that there are multiple kinds of menu's
or 2.  Merge some of the new functions with old ones and rename the other existing functions to avoid confusion.